example/exampleLineCrossSection.js

```diff
@@ -209,23 +209,35 @@ function create_charts(
         }
       }
 
+      let is_cross_section_enabled = false
+      pointMapChart.onDrawEvent(
+        LassoGlobalEventConstants.LASSO_TOOL_TYPE_ACTIVATED,
+        event_obj => {
+          if (event_obj.lasso_tool_type === LassoToolSetTypes.kCrossSection) {
+            is_cross_section_enabled = true
+          }
+        }
+      )
+
+      pointMapChart.onDrawEvent(
+        LassoGlobalEventConstants.LASSO_TOOL_TYPE_DEACTIVATED,
+        event_obj => {
+          if (event_obj.lasso_tool_type === LassoToolSetTypes.kCrossSection) {
+            is_cross_section_enabled = false
+          }
+        }
+      )
+
       // setup an event trigger for when a lasso shape is created
       pointMapChart.onDrawEvent(
         LassoGlobalEventConstants.LASSO_SHAPE_CREATE,
         event_obj => {
           if (cross_section_layer) {
-            const { shape } = event_obj
-            if (shape instanceof LatLonPolyLine) {
-              // if the newly created lasso shape is a 'LatLonPolyLine', which is
-              // the current name of a cross section line shape, then attach an
-              // edit callback to it to capture any/all edits after creation.
-
-              // TODO(croot): find a better way to tag a shape as being a "cross-section line"
-              // Using 'instance of' may be too broad if we end up using LatLonPolyLine as a
-              // general tool for generating a line to filter results using a 'buffer'
-              // Also 'instance of' could be too narrow if we use another shape to draw a cross
-              // section other than LatLonPolyLine
-              // LatLonPolyLine will suffice for now as a differentiator.
+            const { shape, lasso_tool_type = null } = event_obj
+            if (lasso_tool_type === LassoToolSetTypes.kCrossSection) {
```

Comment:
The new `LassoToolSetTypes` enum allows for a much more clean and flexible way to check if the shape event is associated with the cross-section line draw tool. In this case we pull the `lasso_tool_type` attribute, which will be available on most of the `LassoGlobalEventConstants`